device/tt_silicon_driver.cpp

```diff
@@ -531,6 +537,28 @@ void TTDevice::do_open() {
 
         this->system_reg_start_offset = (512 - 16) * 1024*1024;
         this->system_reg_offset_adjust = (512 - 32) * 1024*1024;
+    } else if(is_blackhole(device_info.out)) {
+
+        if (bar2_uc_mapping.mapping_id != TENSTORRENT_MAPPING_RESOURCE2_UC) {
+            throw std::runtime_error(std::string("Device ") + std::to_string(index) + " has no BAR4 UC mapping.");
+        }
+
+        // WC mapping
+        // this->bar4_mapping_size = bar2_wc_mapping.mapping_size;
+        // this->bar4_mapping = mmap(NULL, bar2_wc_mapping.mapping_size, PROT_READ | PROT_WRITE, MAP_SHARED, device_fd, bar2_wc_mapping.mapping_base);
+
+        // if (this->bar4_mapping == MAP_FAILED) {
+        //     throw std::runtime_error(std::string("BAR4 WC memory mapping failed for device ") + std::to_string(index) + ".");
+        // }
+
+        // UC mapping
+        this->bar4_mapping_size = bar2_uc_mapping.mapping_size;
```

Comment:
This looks confusing.  If I understand the code, `bar2_uc_mapping` is misnamed and always corresponds to BAR4, even in WH code path.  `resource2_uc_mapping` or `bar4_uc_mapping` might be a better name?  What do you think?

Reply:
Yeah I see what you are saying, I think the easiest way is to rename bar* variables to proper indexing. I can also add the comment saying that our resource -> bar mapping is (0,1,2) -> (0,2,4). I will see what is the best place to add the comment